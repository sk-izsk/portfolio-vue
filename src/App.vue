<template>
  <div id="app">
    <Home />
  </div>
</template>

<script>
import Home from "./screens/Home";
export default {
  name: "App",
  components: {
    Home,
  },
};
</script>

<style></style>
