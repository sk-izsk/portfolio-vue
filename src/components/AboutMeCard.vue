<template>
  <div>
    <b-card no-body class="overflow-hidden" style="max-width: 540px;">
      <b-row no-gutters>
        <b-col md="6">
          <b-card-img :src="url" alt="Image" class="rounded-0"></b-card-img>
        </b-col>
        <b-col md="6">
          <b-card-body title="Zeeshan">
            <b-card-text>
              I am Shaikh Zeeshan Murshed, Front-End developer from Montreal, Canada. I have experience in making
              Website. Also I am a hobbyist photographer.
            </b-card-text>
            <b-button pill class="btn" @click="openCv" href="#">Download Resume</b-button>
          </b-card-body>
        </b-col>
      </b-row>
    </b-card>
  </div>
</template>

<script>
export default {
  name: 'AboutMeCard',
  props: {
    url: String,
  },
  methods: {
    openCv() {
      window.open(
        'https://docs.google.com/document/d/16Ko4S8n50fedwi4re2K-djKxe2zGrRRWKuXdtko7jow/edit?usp=sharing',
        '_blank',
      );
    },
  },
};
</script>

<style scoped>
.card-img {
  height: 100%;
}
.btn {
  background: linear-gradient(to right, #56ccf2, #2f80ed);
  border-style: none;
  transition: transform 0.5s ease;
}

.btn:hover {
  transform: scale(1.1);
}
</style>
