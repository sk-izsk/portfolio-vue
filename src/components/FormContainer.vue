<template>
  <div class="form-container">
    <b-modal :visible="successModal" title="Send Successfully" centered>
      <b-icon class="icon" variant="primary" icon="hand-thumbs-up"></b-icon>
      <h5 class="my-4">Will get back to you as soon as possible</h5>
    </b-modal>

    <b-modal :visible="errorModal" title="Opps Something goes wrong !!!!!" centered>
      <b-icon class="icon" variant="primary" icon="exclamation-diamond-fill"></b-icon>
      <h5 class="my-4">
        Seems like something went wrong. Kindly <span @click="openEmail" class="text">email</span> me.
      </h5>
    </b-modal>

    <b-form class="form-group-parent" @submit="handleSubmit">
      <b-form-group label="Your Name:">
        <b-form-input v-model="form.name" required placeholder="Enter name"></b-form-input>
      </b-form-group>
      <b-form-group label="Email address:">
        <b-form-input v-model="form.email" type="email" required placeholder="Enter email"></b-form-input>
      </b-form-group>

      <b-form-group label="Subject:">
        <b-form-input v-model="form.subject" required placeholder="Enter subject"></b-form-input>
      </b-form-group>
      <b-form-group label="Message:">
        <b-form-textarea
          placeholder="Enter message"
          required
          rows="3"
          no-resize
          v-model="form.message"
        ></b-form-textarea>
      </b-form-group>
      <div class="btn-container">
        <b-button pill class="btn ml-2" type="submit" size="lg">Send Message</b-button>
      </div>
      <b-card-text class="text-center">
        Don't like forms? Send me an <span @click="openEmail" class="text">Email</span>
      </b-card-text>
    </b-form>
  </div>
</template>

<script>
import { formSchema } from '../validation/validation';
import { sendEmail } from '../api/api';
export default {
  name: 'FormContainer',
  data() {
    return {
      form: {
        email: '',
        name: '',
        subject: '',
        message: '',
      },
      successModal: false,
      errorModal: false,
    };
  },
  methods: {
    openEmail() {
      window.open('mailto:sk.zeeshan1992@gmail.com', '_blank');
    },
    handleSubmit(e) {
      e.preventDefault();
      const { name, email, subject, message } = this.form;
      const payload = {
        name,
        email,
        subject,
        message,
      };
      formSchema.validate(payload).then((payload) => {
        const addEnvironment = {
          ...payload,
          environment: process.env.NODE_ENV === 'development' ? 'development' : 'production',
        };
        this.$emit('sendingEmail', true);
        sendEmail(addEnvironment)
          .then((response) => {
            if (response.status === 200) {
              this.form = {
                email: '',
                name: '',
                subject: '',
                message: '',
              };
              this.$emit('sendingEmail', false);
              this.successModal = true;
            }
          })
          .catch((err) => {
            this.errorModal = true;
            console.warn(err);
          });
      });
    },
  },
};
</script>

<style scoped>
.form-container {
  display: flex;
  justify-content: center;
}
.form-group-parent {
  width: 80%;
}
.btn-container {
  width: 100%;
  display: flex;
  justify-content: center;
}
.btn {
  background: linear-gradient(to right, #56ccf2, #2f80ed);
  border-style: none;
  transition: transform 0.5s ease;
}
.btn:hover {
  transform: scale(1.1);
  background: linear-gradient(to right, #2f80ed, #56ccf2);
}

.text {
  margin-left: 5px;
  cursor: pointer;
  font-weight: 500;
  color: #2f80ed;
}
.center {
  text-align: center !important;
}

.icon {
  height: 30px;
  width: 30px;
  cursor: pointer;
}
.text-center {
  margin-top: 4px;
}
.modal-body {
  display: flex !important;
  flex-direction: column !important;
  align-items: center !important;
}
</style>
