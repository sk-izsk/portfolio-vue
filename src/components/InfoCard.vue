<template>
  <div>
    <div :key="information.nameOfOrganization" class="info-card" v-for="information in informations">
      <b-icon class="icon" :icon="iconName" variant="primary"></b-icon>
      <b-card class="card-container" :title="information.nameOfOrganization" :sub-title="(information.startYear)">
        <b-card-text>
          {{ information.details }}
        </b-card-text>
      </b-card>
    </div>
  </div>
</template>

<script>
import { BCard, BCardText, BIcon } from 'bootstrap-vue';
export default {
  name: 'InfoCard',
  components: {
    BCard,
    BCardText,
    BIcon,
  },
  props: {
    informations: Array,
    iconName: String,
  },
};
</script>

<style scoped>
.info-card {
  padding: 0px 20px;
}
.icon {
  color: white;
  width: 30px;
  height: 30px;
  /* margin-top: 30px; */
  position: relative;
  top: 54px;
  z-index: 1;
  left: 7px;
}
.icon:hover {
  color: #ffd15c;
}

.card-container {
  display: flex;
  flex-direction: row;
  padding-left: 20px;
}
</style>
