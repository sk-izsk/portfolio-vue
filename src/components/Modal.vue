<template>
  <div>
    <b-modal :visible="showModal" :title="title">
      <b-icon class="icon" variant="primary" :icon="iconName"></b-icon>
      <h5 class="my-4">{{ description }}</h5>
    </b-modal>
  </div>
</template>

<script>
export default {
  name: 'Modal',
  props: {
    description: String,
    showModal: Boolean,
    title: String,
    iconName: String,
  },
};
</script>

<style scoped></style>
