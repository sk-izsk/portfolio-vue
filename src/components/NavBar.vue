<template>
  <div>
    <b-navbar :sticky="true" toggleable="lg" type="dark" variant="dark">
      <b-navbar-brand href="#">iZsk</b-navbar-brand>
      <b-navbar-toggle target="nav-collapse"></b-navbar-toggle>
      <b-collapse id="nav-collapse" is-nav>
        <b-navbar-nav>
          <b-nav-item href="#home">Home</b-nav-item>
          <b-nav-item href="#about">About</b-nav-item>
          <b-nav-item href="#">Education</b-nav-item>
        </b-navbar-nav>
        <b-navbar-nav class="ml-auto">
          <b-nav-item href="#">Experience</b-nav-item>
          <b-nav-item href="#">Photography</b-nav-item>
          <b-nav-item href="#">Contact</b-nav-item>
        </b-navbar-nav>
      </b-collapse>
    </b-navbar>
  </div>
</template>

<script>
export default {
  name: 'NaveBar',
};
</script>

<style scoped></style>
