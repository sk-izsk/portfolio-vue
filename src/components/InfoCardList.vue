<template>
  <div>
    <h1>{{ title }}</h1>
    <InfoCard :iconName="iconName" :informations="informations" />
  </div>
</template>

<script>
import InfoCard from './InfoCard';
export default {
  name: 'InfoCardList',
  components: {
    InfoCard,
  },
  props: {
    title: String,
    informations: Array,
    iconName: String,
  },
};
</script>

<style scoped>
h1 {
  text-align: center;
}
</style>
