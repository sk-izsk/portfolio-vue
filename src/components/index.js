import AboutMeCard from './AboutMeCard';
import FormContainer from './FormContainer';
import InfoCard from './InfoCard';
import InfoCardList from './InfoCardList';
import Modal from './Modal';
import NavBar from './NavBar';
import ProgressContainer from './ProgressContainer';
import Spinner from './Spinner';

export { NavBar, AboutMeCard, ProgressContainer, Spinner, InfoCard, InfoCardList, FormContainer, Modal };
