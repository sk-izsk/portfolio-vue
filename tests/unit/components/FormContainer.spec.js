import { shallowMount } from '@vue/test-utils';
import { FormContainer } from '../../../src/components/';

describe('FormContainer', () => {
  it('renders when passed', () => {
    const wrapper = shallowMount(FormContainer);
    expect(wrapper).toMatchSnapshot();
  });
});
