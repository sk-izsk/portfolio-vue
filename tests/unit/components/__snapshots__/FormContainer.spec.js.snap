// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormContainer renders when passed 1`] = `
<div class="form-container">
  <b-modal-stub size="md" centered="true" ignoreenforcefocusselector="" title="Send Successfully" titletag="h5" headerclosecontent="&amp;times;" headercloselabel="Close" canceltitle="Cancel" oktitle="OK" cancelvariant="secondary" okvariant="primary">
    <b-icon-stub icon="hand-thumbs-up" variant="primary" fontscale="1" scale="1" rotate="0" shifth="0" shiftv="0" class="icon"></b-icon-stub>
    <h5 class="my-4">Will get back to you as soon as possible</h5>
  </b-modal-stub>
  <b-modal-stub size="md" centered="true" ignoreenforcefocusselector="" title="Opps Something goes wrong !!!!!" titletag="h5" headerclosecontent="&amp;times;" headercloselabel="Close" canceltitle="Cancel" oktitle="OK" cancelvariant="secondary" okvariant="primary">
    <b-icon-stub icon="exclamation-diamond-fill" variant="primary" fontscale="1" scale="1" rotate="0" shifth="0" shiftv="0" class="icon"></b-icon-stub>
    <h5 class="my-4">
      Seems like something went wrong. Kindly <span class="text">email</span> me.
    </h5>
  </b-modal-stub>
  <b-form-stub class="form-group-parent">
    <b-form-group-stub label="Your Name:" feedbackarialive="assertive">
      <b-form-input-stub required="true" value="" placeholder="Enter name" debounce="0" type="text"></b-form-input-stub>
    </b-form-group-stub>
    <b-form-group-stub label="Email address:" feedbackarialive="assertive">
      <b-form-input-stub required="true" value="" placeholder="Enter email" debounce="0" type="email"></b-form-input-stub>
    </b-form-group-stub>
    <b-form-group-stub label="Subject:" feedbackarialive="assertive">
      <b-form-input-stub required="true" value="" placeholder="Enter subject" debounce="0" type="text"></b-form-input-stub>
    </b-form-group-stub>
    <b-form-group-stub label="Message:" feedbackarialive="assertive">
      <b-form-textarea-stub required="true" value="" placeholder="Enter message" debounce="0" rows="3" wrap="soft" noresize="true"></b-form-textarea-stub>
    </b-form-group-stub>
    <div class="btn-container">
      <b-button-stub size="lg" variant="secondary" type="submit" tag="button" pill="true" target="_self" class="btn ml-2">Send Message</b-button-stub>
    </div>
    <b-card-text-stub texttag="p" class="text-center">
      Don't like forms? Send me an <span class="text">Email</span></b-card-text-stub>
  </b-form-stub>
</div>
`;
